web/src/app/core/resolvers/ping.resolver.ts

```
@@ -0,0 +1,36 @@
+// Angular modules
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Resolve, Router } from '@angular/router';
+
+// Third party modules
+import { of } from 'rxjs';
+import { catchError, switchMap, take } from 'rxjs/operators';
+import { PingModel } from '../../shared/models/index.model';
+import { PingService } from '../services/ping.service';
+
+// Dashboard hub model and services
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PingResolver implements Resolve<PingModel> {
+
+  constructor(
+    private pingService: PingService,
+    private router: Router
+  ) { }
+
+  resolve(route: ActivatedRouteSnapshot): any {
+    return this.pingService.findOneById(route.params.projectUid, route.params.monitorUid)
+      .pipe(
+        take(1),
+        switchMap((pings: PingModel[]) => {
+          return of(pings);
+        }),
+        catchError(() => {
```

This is hiding the error...

```
core.js:15724 ERROR Error: Uncaught (in promise): FirebaseError: [code=permission-denied]: Missing or insufficient permissions.
FirebaseError: Missing or insufficient permissions.
```